### PR TITLE
Refactoring store liveness update

### PR DESCRIPTION
**Pull request by BrianBohe**
Following up [79182](https://github.com/dotnet/runtime/pull/79182). When storing a variable means a spill, we are updating its liveness and setting the register to REG_STK. When the target is a register and not the stack, then we usually call genProduceReg which triggers the liveness update. This pr refactor similar code to avoid unintentional changes breaking either var home or var liveness and fix one case on arm 64 which was not considered (emitIns_S_R and emitIns_R_I don't update var home).

### CODE REVIEW

**Comment by BrianBohe** [comment]
cc @dotnet/jit-contrib

**Comment by BrianBohe** [comment]
@kunalspathak are you able to take a look at this pr? All errors seem to be known issues.

**Comment by kunalspathak** [comment]
Seems like this affects the TP a bit. Can you please verify to see where it is coming from? Perhaps mark the method `genUpdateLifeStore()` with `forceinline`? 

image

File: src/coreclr/jit/codegen.h [hunk 1]
```diff
@@ -1193,8 +1193,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 #endif // !defined(TARGET_64BIT)
 
-    // Do liveness update for register produced by the current node in codegen after
```
Comment (kunalspathak) [inline src/coreclr/jit/codegen.h:1196]: Can you keep the comment?

Comment (BrianBohe) [reply src/coreclr/jit/codegen.h:1029]: Thanks!

**Comment by BrianBohe** [comment]
There is no difference on TP when using `__forceinline`, either compiling with `/p:NoPgoOptimize=true` or without.

**Comment by kunalspathak** [comment]
Can you merge `main` to get the fix for https://github.com/dotnet/runtime/issues/82506 and check the superpmi jobs?

**Comment by kunalspathak** [comment]
Changes looks good to me. Once we get the superpmi numbers, we can go ahead and merge.

**Comment by BrianBohe** [comment]
> Changes looks good to me. Once we get the superpmi numbers, we can go ahead and merge.

TP is overall positive:
image

There is a timeout in OSX-x64 which seems to be non-related

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**